test/e2e/performanceprofile/functests/2_performance_update/tuned_deferred.go

```diff
@@ -0,0 +1,320 @@
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/discovery"
+	testlog "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/log"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/mcps"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/nodes"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/pods"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/profiles"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/profilesupdate"
+)
+
+var _ = Describe("Tuned Deferred tests of performance profile", Ordered, Label(string("Tier5")), func() {
+	var (
+		workerCNFNodes []corev1.Node
+		err            error
+		poolName       string
+		np             *hypershiftv1beta1.NodePool
+		tuned          *tunedv1.Tuned
+	)
+
+	type TunedProfileConfig struct {
+		DeferMode         string
+		ProfileChangeType string // "first-time" or "in-place"
+		ExpectedBehavior  string // "deferred" or "immediate"
+		Kernel_shmmni     string // 4096 or 8192
+	}
+
+	name := "performance-patch"
+
+	BeforeEach(func() {
+		if discovery.Enabled() && testutils.ProfileNotFound {
+			Skip("Discovery mode enabled, performance profile not found")
+		}
+		workerCNFNodes, err = nodes.GetByLabels(testutils.NodeSelectorLabels)
+		Expect(err).ToNot(HaveOccurred())
+		workerCNFNodes, err = nodes.MatchingOptionalSelector(workerCNFNodes)
+		Expect(err).ToNot(HaveOccurred(), "error looking for the optional selector: %v", err)
+		Expect(workerCNFNodes).ToNot(BeEmpty())
+
+		profile, err := profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+		Expect(err).ToNot(HaveOccurred())
+
+		poolName, err = mcps.GetByProfile(profile)
+		Expect(err).ToNot(HaveOccurred())
+		testlog.Infof("using performanceMCP: %q", poolName)
+	})
+
+	Context("Tuned Deferred status", func() {
+		DescribeTable("Validate Tuned DeferMode behavior",
+			func(tc TunedProfileConfig) {
+
+				if tc.ProfileChangeType == "first-time" {
+					tuned = getTunedProfile(name)
+
+					// If profile is present
+					if tuned != nil {
+
+						// Delete the profile
+						testlog.Infof("Deleting Tuned Profile Patch: %s", name)
+						Expect(testclient.ControlPlaneClient.Delete(context.TODO(), tuned)).To(Succeed())
+						time.Sleep(2 * time.Minute)
+					}
+					tuned, err = createTunedObject(tc.DeferMode, tc.Kernel_shmmni, np, name)
+					Expect(err).NotTo(HaveOccurred())
+
+				} else {
+					tuned := getTunedProfile(name)
+					tunedData := []byte(*tuned.Spec.Profile[0].Data)
+
+					cfg, err := ini.Load(tunedData)
+					Expect(err).ToNot(HaveOccurred())
+
+					sysctlSection, err := cfg.GetSection("sysctl")
+					Expect(err).ToNot(HaveOccurred())
+
+					sysctlSection.Key("kernel.shmmni").SetValue(tc.Kernel_shmmni)
```

Comment:
the fact the value is bound to a specific key should be more evident in the test definition IMO (can be cleaned up later)